### PR TITLE
Fix/update challenges for basic html and html5

### LINKED ISSUE #18007: Ordered and Unordered list can be empty
Probably not a big deal, but in the ordered and unordered list session the tests pass even if the list items are empty

https://learn.freecodecamp.org/responsive-web-design/basic-html-and-html5/create-a-bulleted-unordered-list

* Browser Name: Chrome
* Browser Version: 69.0.3497.32 (Official Build) beta (64-bit)
* Operating System: Mac OS High Seirra

screen shot 2018-08-12 at 11 25 26

### PR DESCRIPTION
**Pull request by aditijoshi14**
Currently, for this following challenge:
https://learn.freecodecamp.org/responsive-web-design/basic-html-and-html5/create-a-bulleted-unordered-list/
An empty string was being accepted. With this fix, it will not allow an empty string or whitespaces as an input. 

- [X] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `master` branch of freeCodeCamp.
- [X] None of my changes are plagiarized from another source without proper attribution.
- [X] All the files I changed are in the same world language (only English)
- [X] My changes do not use shortened URLs or affiliate links.

Closes part of issue #18007

### CODE REVIEW

**Comment by RandellDawson** [comment]
@aditijoshi14 You are probably wondering why you have a merge conflict.  You probably need to run:
`git pull --rebase upstream master` on your local branch with the changes and then resolve the conflict if it still exists.

**Comment by aditijoshi14** [comment]
@RandellDawson Thank you so much I was able to resolve the conflict.

**Comment by RandellDawson** [comment]
I don't think you meant to have that other file on the PR did you?

**Comment by aditijoshi14** [comment]
Nope. I didn't mean to have the other file on this PR but I had to commit that to resolve the conflict. I apologize for the confusion.

File: curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/create-a-bulleted-unordered-list.english.md [hunk 1]
```diff
@@ -32,7 +32,8 @@ tests:
     testString: assert(code.match(/<\/ul>/gi) && code.match(/<ul/gi) && code.match(/<\/ul>/gi).length === code.match(/<ul/gi).length, 'Make sure your <code>ul</code> element has a closing tag.');
   - text: Make sure your <code>li</code> elements have closing tags.
     testString: assert(code.match(/<\/li>/gi) && code.match(/<li[\s>]/gi) && code.match(/<\/li>/gi).length === code.match(/<li[\s>]/gi).length, 'Make sure your <code>li</code> elements have closing tags.');
-
+  - text: Make sure your <code>li</code> elements doesn’t contain empty string or only whitespace.
+    testString: assert($("ul li").filter(function(index){return $(this).text().trim().length === 0;}).length === 0, 'Make sure your <code>li</code> elements doesn\’t contain empty string or only whitespace.');
```
Comment (RandellDawson) [inline curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/create-a-bulleted-unordered-list.english.md]: While your added test works fine, yo could use the same logic but simplify a bit.
```suggestion
    testString: assert($("ul li").filter((_, item) => !$(item).text().trim()).length === 0, 'Make sure your <code>li</code> elements don\’t contain an empty string or only white-space.');
```

File: curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/create-a-bulleted-unordered-list.english.md [hunk 2]
```diff
@@ -32,7 +32,8 @@ tests:
     testString: assert(code.match(/<\/ul>/gi) && code.match(/<ul/gi) && code.match(/<\/ul>/gi).length === code.match(/<ul/gi).length, 'Make sure your <code>ul</code> element has a closing tag.');
   - text: Make sure your <code>li</code> elements have closing tags.
     testString: assert(code.match(/<\/li>/gi) && code.match(/<li[\s>]/gi) && code.match(/<\/li>/gi).length === code.match(/<li[\s>]/gi).length, 'Make sure your <code>li</code> elements have closing tags.');
-
+  - text: Make sure your <code>li</code> elements doesn’t contain empty string or only whitespace.
```
Comment (RandellDawson) [inline curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/create-a-bulleted-unordered-list.english.md]: Please make the following changes to the text.
```suggestion
  - text: Make sure your <code>li</code> elements don’t contain an empty string or only white-space.
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
Please see my requested changes.

**Comment by RandellDawson** [comment]
Please see my suggestions above and remove the changes to that extra file from this PR.

Thank you.

**Comment by RandellDawson** [comment]
You still have the other file on this PR.  In the future, before you start making changes, do a rebase on your master branch `git pull --rebase upstream master` and then checkout a new branch (`git checkout -b branchname`) before making changes for your next PR.  It appears this PR has some changes left over from another branch you made for another PR.

**Comment by RandellDawson** [comment]
That is why your commit history for this PR goes all the way back to December 6, 2018.

**Comment by aditijoshi14** [comment]
I apologize for that. I will surely do that from next time. Do you happen to know if there is anyway I can delete that extra file from this PR?

**Comment by aditijoshi14** [comment]
If you want I can create a new branch, add changes into it and make a new pull request.

**Comment by RandellDawson** [comment]
No, just keep this PR open and I will try to help you out.  It is a little more complicated, because you have made several other commits trying to resolve this and I get merge conflicts when I try to rebase your PR locally.  In the end it will be a good learning experience for both of us.  I don't have time to debug this now, but will reply later when I have some more time to work on this.

**Comment by RandellDawson** [comment]
What happens if you `git pull --rebase upstream master on your local `ix/update-challenges-for-basic-html-and-html5` branch?  Do you get a merge conflict?

**Comment by aditijoshi14** [comment]
Yes, I get a marge conflict.

**Comment by aditijoshi14** [comment]
@RandellDawson  I was able to resolve the merge conflict and remove the extra file. Thank you so much for all your help!

**Comment by RandellDawson** [comment]
Just out of curiosity, how did you finally remove it?  I figured out two ways to do it, but you beat me to it.

**Comment by RandellDawson** [comment]
Can you visit the Contributer's channel?  I need to discuss something with you.

**Comment by aditijoshi14** [comment]
@RandellDawson  I actually synced my forked repo with FreeCodeCamp's repo, which I believe was giving me merge conflict in my third commit. My third commit was when I deleted the extra file from my PR directly. After syncing my forked repo, I reverted to the third commit and made the changes you suggested. 

I am currently in Contributor's channel in Gitter.

**Review by RandellDawson — APPROVED** [review]

**Review by RandellDawson — APPROVED** [review]

**Comment by RandellDawson** [comment]
I have tested the changes locally and validate the new test added works as intended without any "side-effects" that I can see.

**Review by ezioda004 — APPROVED** [review]

**Comment by ezioda004** [comment]
Thank you @aditijoshi14 for your contribution and congratulations on your first PR merge to this repo! 🎉
Hope to see more contributions from you in the future.

**Pull request merged**